glib-macros/src/utils.rs

```diff
@@ -214,6 +214,12 @@ pub fn crate_ident_new() -> TokenStream {
     })
 }
 
+#[cfg(not(feature = "proc_macro_crate"))]
+pub fn crate_ident_new() -> TokenStream {
+    let glib = Ident::new("glib", Span::call_site());
+    quote!(#glib)
```

Comment:
This will break the macros when used from `glib` (so `glib` needs to enable the feature which makes this change rather pointless), and also in various other situations (renaming dependencies, ...)

Reply:
why would it break if the crate name is glib? it doesn't break for glib build itself, what do you mean?

Reply:
You can't reference `glib` as `glib` from inside itself. The tests should be failing if this feature is disabled (because nothing other than tests currently uses `glib-macros`).

Reply:
I think glib tests use "glib" name too. otherwise some trick like "extern crate self as glib;" should help, but I apparently not needed.